flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveOptions.java

```diff
@@ -134,6 +134,17 @@ public class HiveOptions {
     public static final ConfigOption<String> SINK_PARTITION_COMMIT_SUCCESS_FILE_NAME =
             FileSystemConnectorOptions.SINK_PARTITION_COMMIT_SUCCESS_FILE_NAME;
 
+    public static final ConfigOption<Boolean> TABLE_EXEC_HIVE_SINK_STATISTIC_AUTO_GATHER_ENABLE =
+            key("table.exec.hive.sink.statistic-auto-gather.enable")
+                    .booleanType()
+                    .defaultValue(false)
```

Comment:
What's the default behavior of Hive/Spark? Should we enable it by default? The streaming mode can ignore this configuration by default.

Reply:
Hive enable it by default, but spark disabed it by default. The reason I disabled it by default is for gathering statistic may be time-consuming, and user may find his job is slower when upgrade to Flink1.16 if they haven't noticed Flink will auto gather statistic since 1.16.
But after discussion with @Tartarus0zm.  I changes my mind, I think we can just follow Hive's default behavior since we want to provide compatibility to Hive .
Also, I'll note enable auto gather statistic may spend some time in the document.